group-coordinator/src/main/java/org/apache/kafka/coordinator/group/taskassignor/TopologyDescriber.java

```diff
@@ -34,4 +37,8 @@ public interface TopologyDescriber {
      */
     int numPartitions(String subtopologyId);
 
+    default Set<Integer> statefulTaskIds(String subtopologyId) {
+        return new HashSet<>();
```

Comment:
Either all tasks are stateful or none are. Could we just make this a 
`boolean isStateful(String subtopologyId)` and then build the task IDs using `numPartitions`?

It would also be nice to move the implementation to the implementation class. If you can implement it already that would be great. If not, please leave a TODO so that we remember to implement this.